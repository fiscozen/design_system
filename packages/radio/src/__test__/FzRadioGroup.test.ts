import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";
import { h } from "vue";
import { FzRadioGroupProps, FzRadioGroup, FzRadio } from "..";

const createWrapper = async (props: FzRadioGroupProps) => {
  const content = mount(FzRadioGroup, {
    props,
    slots: {
      default: (props) => [
        h(FzRadio, {
          label: "Radio 1",
          value: "Radio 1",
          ...props.radioGroupProps,
        }),
        h(FzRadio, {
          label: "Radio 2",
          value: "Radio 2",
          ...props.radioGroupProps,
        }),
      ],
    },
  });
  return content;
};

describe("FzRadioGroup", () => {
  it("should render correctly", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
      name: "radio",
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should set error state", async () => {
    const wrapper = mount(FzRadioGroup, {
      props: {
        label: "Radio",
        size: "md",
        error: true,
      },
      slots: {
        default: (props) => [
          h(FzRadio, {
            label: "Radio 1",
            value: "Radio 1",
            ...props.radioGroupProps,
          }),
          h(FzRadio, {
            label: "Radio 2",
            value: "Radio 2",
            ...props.radioGroupProps,
          }),
        ],
        error: "Error text",
      },
    });

    await wrapper.vm.$nextTick();

    expect(wrapper.find("p")).toBeTruthy();
    expect(wrapper.find("p").text()).toBe("Error text");
    expect(wrapper.find("p").find("svg")).toBeTruthy();
    wrapper
      .find('[test-id="slot-container"]')
      .findAll("label")
      .forEach((input) => {
        expect(input.classes()).toContain("before:border-semantic-error");
      });
  });

  it("should set disabled state", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
      disabled: true,
    });

    await wrapper.vm.$nextTick();
    wrapper.findAll("input").forEach((input) => {
      expect(input.element.disabled).toBe(true);
    });
  });

  it("should set emphasized class", async () => {
    const wrapper = await createWrapper({
      label: "Radio",
      size: "md",
      emphasis: true,
    });
    await wrapper.vm.$nextTick();
    wrapper
      .find('[test-id="slot-container"]')
      .findAll("label")
      .forEach((input) => {
        expect(input.classes()).toContain("before:border-blue-500");
      });
  });
});
