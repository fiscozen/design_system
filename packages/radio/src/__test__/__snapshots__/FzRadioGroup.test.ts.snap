// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FzRadioGroup > should render correctly 1`] = `
"<div class="flex flex-col text-md gap-12"><label for="radio" class="flex flex-col text-md gap-6 text-core-black"><span>Radio<!--v-if--></span>
    <!--v-if-->
  </label>
  <div class="flex flex-col text-md gap-8" test-id="slot-container">
    <div data-v-029ccdb9="" class="flex justify-center flex-col w-fit"><input data-v-029ccdb9="" type="radio" id="Radio 1" class="radio--medium peer" name="radio" tabindex="0" value="Radio 1"><label data-v-029ccdb9="" for="Radio 1" class="flex items-start gap-4 h-16 
  before:content-[''] before:inline-block before:border-solid before:border-1 before:rounded-full before:ml-4
  peer-checked:before:bg-transparent peer-focus:before:outline peer-checked:before:outline-offset-1 peer-checked:before:outline-1 peer-checked:before:outline-blue-600
  peer-disabled:text-grey-300
  peer-disabled:before:border-grey-200 peer-disabled:before:bg-grey-200
  peer-checked:peer-disabled:before:border-grey-200 text-md before:h-16 before:w-16 before:mt-4 peer-checked:before:border-[5px] before:border-grey-500"><span data-v-029ccdb9="" class="w-fit">Radio 1</span></label>
      <!--v-if-->
    </div>
    <div data-v-029ccdb9="" class="flex justify-center flex-col w-fit"><input data-v-029ccdb9="" type="radio" id="Radio 2" class="radio--medium peer" name="radio" tabindex="0" value="Radio 2"><label data-v-029ccdb9="" for="Radio 2" class="flex items-start gap-4 h-16 
  before:content-[''] before:inline-block before:border-solid before:border-1 before:rounded-full before:ml-4
  peer-checked:before:bg-transparent peer-focus:before:outline peer-checked:before:outline-offset-1 peer-checked:before:outline-1 peer-checked:before:outline-blue-600
  peer-disabled:text-grey-300
  peer-disabled:before:border-grey-200 peer-disabled:before:bg-grey-200
  peer-checked:peer-disabled:before:border-grey-200 text-md before:h-16 before:w-16 before:mt-4 peer-checked:before:border-[5px] before:border-grey-500"><span data-v-029ccdb9="" class="w-fit">Radio 2</span></label>
      <!--v-if-->
    </div>
  </div>
  <!--v-if-->
</div>"
`;
